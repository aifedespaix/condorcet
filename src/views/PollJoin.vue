<template>
  <div class="Container">
    <div>{{ message }}</div>
  </div>
</template>

<script>
import { mapGetters } from "vuex";
import { db } from "../firebase";

export default {
  name: "PollJoin",
  data() {
    return {
      message: "Traitement en cours..."
    };
  },
  computed: {
    ...mapGetters("user", ["userId"])
  },
  async mounted() {
    try {
      const { params, query } = this.$route;
      const document = db.collection("polls").doc(params.id);
      const poll = (await document.get()).data();
      const { users, token: pollToken } = poll;

      if (!query.token) {
        return;
      }

      if (query.token !== pollToken) {
        return;
      }

      const newPoll = { ...poll, users: [...new Set([...users, this.userId])] };

      document.set(newPoll);

      this.message = newPoll.question;
      setTimeout(() => {
        if (newPoll.isActive) {
          this.$router.push({ name: "poll_show", params: params.id });
          return;
        }

        this.$router.push({ name: "poll_result", params: params.id });
      }, 2000);
    } catch (error) {
      console.error(error);
    }
  }
};
</script>

<style>
.Container {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100vh;
}
</style>
