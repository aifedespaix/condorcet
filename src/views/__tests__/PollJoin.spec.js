import { shallowMount } from "@vue/test-utils";
import PollJoin from "../PollJoin.vue";

jest.mock("../../firebase", () => ({
  db: {
    collection: () => ({
      doc: () => ({
        get: () => ({
          data: () => ({
            token: "le-token"
          })
        })
      })
    })
  }
}));

const $route = {
  params: "adz93dze823j",
  query: { q: "le-token" }
};

describe("PollJoin.vue", () => {
  it("should render", () => {
    const wrapper = shallowMount(PollJoin, { mocks: { $route } });

    expect(wrapper).toMatchSnapshot();
  });
});
