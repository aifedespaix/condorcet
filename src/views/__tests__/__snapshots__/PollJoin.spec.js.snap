// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollJoin.vue should render 1`] = `
<div class="Container">
  <div>Qui est le meilleur ?</div>
</div>
`;
